File: src/main/java/net/jeikobu/jbase/command/Command.java
package net.jeikobu.jbase.command;

import sx.blah.discord.handle.obj.Permissions;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface Command {
    String name();
    int argsLength() default 0;
    Permissions[] permissions() default Permissions.SEND_MESSAGES;
}

File: src/main/java/net/jeikobu/jbase/command/Command.kt
package net.jeikobu.jbase.command

import net.dv8tion.jda.core.Permission

@Retention(AnnotationRetention.RUNTIME)
annotation class Command(val name: String, val aliases: Array<String> = [],
                         val argsLength: Int = 0,  val permissions: Array<Permission> = [])